addons/portal/wizard/portal_share.py

```diff
@@ -53,23 +52,29 @@ def action_send_mail(self):
         # if partner already user or record has access token send common link in batch to all user
         for partner in self.partner_ids:
             share_link = active_record.get_base_url() + active_record._get_share_url(redirect=True, pid=partner.id)
+            context = dict(lang=partner.lang)
+            template = self.env.ref('portal.portal_share_template', False).with_context(lang=partner.lang)
```

Comment:
if somehow `portal_share_template` is deleted this code will traceback
As the lang may also be found inspecting `self` context
https://github.com/odoo/odoo/blob/b738068be53621b23d9f19befdfdef50534ef434/odoo/tools/translate.py#L421-L424
you could have something like
```diff
         # if partner already user or record has access token send common link in batch to all user
         for partner in self.partner_ids:
             share_link = active_record.get_base_url() + active_record._get_share_url(redirect=True, pid=partner.id)
+            saved_lang = self.env.lang
+            self = self.with_context(lang=partner.lang)
             template = self.env.ref('portal.portal_share_template', False)
             active_record.with_context(mail_post_autofollow=True).message_post_with_view(template,
                 values={'partner': partner, 'note': self.note, 'record': active_record,
                         'share_link': share_link},
                 subject=_("You are invited to access %s") % active_record.display_name,
                 subtype_id=note.id,
                 notif_layout='mail.mail_notification_light',
                 partner_ids=[(6, 0, partner.ids)])
+            self = self.with_context(lang=saved_lang)
```
which seems to work for each content to translate